pkg/channel/message_dispatcher.go

```diff
@@ -239,30 +239,28 @@ func (d *MessageDispatcherImpl) executeRequest(ctx context.Context,
 	}
 	execInfo.Time = dispatchTime
 
-	body := make([]byte, attributes.KnativeErrorDataExtensionMaxLength)
+	body := new(bytes.Buffer)
+	_, readErr := body.ReadFrom(response.Body)
 
 	if isFailure(response.StatusCode) {
 		// Read response body into execInfo for failures
-		readLen, err := response.Body.Read(body)
-		if err != nil && err != io.EOF {
-			d.logger.Error("failed to read response body into DispatchExecutionInfo", zap.Error(err))
+		if readErr != nil && readErr != io.EOF {
+			d.logger.Error("failed to read response body", zap.Error(err))
 			execInfo.ResponseBody = []byte(fmt.Sprintf("dispatch error: %s", err.Error()))
 		} else {
-			execInfo.ResponseBody = body[:readLen]
+			execInfo.ResponseBody = body.Bytes()
```

Comment:
Do you still want to truncate the payload here in error cases?

(I suspect the right thing to do may actually be to truncate the _overall_ message, preferring to cut payload over other attributes.)

Reply:
We don't check if the response is a cloud event in error cases. The payload gets truncated in a later step by a transformer and then put in as a cloud event extension on the original message. It's then sent to the dead letter. So I think the truncation still happens, it's just no longer part of the `executeRequest`

I'll double check the above and add a conformance test

Reply:
> I'll double check the above and add a conformance test

@gab-satchi  Did you add some conformance test for this?

Reply:
@matzew, Just an e2e test right now. I can add one under  [here](https://github.com/knative/eventing/blob/main/test/conformance/helpers/broker_data_plane_test_helper.go#L94) but if we're looking for one that could be reused by other implementations then that seems like a bigger effort. 

I'm seeing all the delivery based ones just TODO-ed [here](https://github.com/knative/eventing/blob/main/test/rekt/features/broker/data_plane.go#L100)